### PR TITLE
DEV-2943 Remove sample info from WgsMetrics and BAM and CRAM

### PR DESCRIPTION
**Pull request by stefvanlieshout**
Remove all sample related info from the contents of output files (of diagnostic pipeline runs) that are re-used in research pipeline runs so that no such information "leaks" into the research context. This will allow to run diagnostic pipeline with different sample name.

- Removes sample info from ReadGroups
- Removes all PG lines from BAM header
- Removes all PG lines from CRAM header
- Removes all non-data content from WgsMetrics output

Note: the reheadering of BAM requires double the disk size in MergeMarkDup step because the output (from samtools reheader) is written to a new file (unfortunately sambamba does not allow for streaming output), so we might need to increase disk size somewhere?

### CODE REVIEW

File: cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/GrepFileCommand.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.hartwig.pipeline.execution.vm.unix;
+
+import static java.lang.String.format;
+
+import com.hartwig.pipeline.execution.vm.BashCommand;
+
+public class GrepFileCommand implements BashCommand {
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/GrepFileCommand.java:7]: GrepFileCommand could extend GrepCommand

File: cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/RedirectStdoutCommand.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.hartwig.pipeline.execution.vm.unix;
+
+import static java.lang.String.format;
+
+import com.hartwig.pipeline.execution.vm.BashCommand;
+
+public class RedirectStdoutCommand implements BashCommand {
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/RedirectStdoutCommand.java:7]: Can we add a unit test for this guy?

File: pom.xml [hunk 3]
```diff
@@ -70,10 +70,21 @@
                 <artifactId>events-pipeline</artifactId>
                 <version>${events.version}</version>
             </dependency>
+<!--            <dependency>-->
```
Comment (pauldwolfe) [inline pom.xml:73]: We can remove these as they didn't have an impact

**Review by pauldwolfe — CHANGES_REQUESTED** [review]

File: cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/GrepFileCommand.java (hunk 1 above)
Comment (stefvanlieshout) [reply cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/GrepFileCommand.java:7]: GrepCommand is no longer needed so removed.

**Review by pauldwolfe — APPROVED** [review]

File: cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/PipeCommands.java [hunk 4]
```diff
@@ -6,7 +6,8 @@
 
 import com.hartwig.pipeline.execution.vm.BashCommand;
 
-public class PipeCommands implements BashCommand {
+public class
+PipeCommands implements BashCommand {
```
Comment (nedleitch) [inline cluster/src/main/java/com/hartwig/pipeline/execution/vm/unix/PipeCommands.java:10]: Looks like a mistake?

File: cluster/src/test/java/com/hartwig/pipeline/execution/vm/unix/DeleteFilesCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.hartwig.pipeline.execution.vm.DeleteFilesCommand;
+
+import org.junit.Test;
+
+public class DeleteFilesCommandTest {
+    @Test
+    public void createsValidCommand() {
+        List<String> filesToDelete = Arrays.asList("file1", "file2");
+        assertThat(new DeleteFilesCommand(filesToDelete).asBash()).isEqualTo("rm file1 file2");
```
Comment (nedleitch) [inline cluster/src/test/java/com/hartwig/pipeline/execution/vm/unix/DeleteFilesCommandTest.java:16]: Inline?

File: cluster/src/test/java/com/hartwig/pipeline/execution/vm/unix/RedirectStdoutCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import com.hartwig.pipeline.calling.command.SamtoolsCommand;
+
+import org.junit.Test;
+
+public class RedirectStdoutCommandTest {
+    @Test
+    public void createsValidCommand() {
+        String grepString = "view file.bam";
+        String outputFile = "/some/output/file";
+        assertThat(new RedirectStdoutCommand(new SamtoolsCommand(grepString), outputFile).asBash()).isEqualTo(format("(/opt/tools/samtools/1.14/samtools %s 1> %s)",
```
Comment (nedleitch) [inline cluster/src/test/java/com/hartwig/pipeline/execution/vm/unix/RedirectStdoutCommandTest.java:16]: I do find this a bit confusing, and it introduces a dependency on `SamtoolsCommand`. Could we simplify it to just test the functionality the class under test provides?

**Review by nedleitch — CHANGES_REQUESTED** [review]

File: cluster/src/test/java/com/hartwig/pipeline/execution/vm/unix/RedirectStdoutCommandTest.java (hunk 6 above)
Comment (stefvanlieshout) [reply cluster/src/test/java/com/hartwig/pipeline/execution/vm/unix/RedirectStdoutCommandTest.java:16]: Agreed and will replace with mock of BashCommand instead.

**Review by nedleitch — APPROVED** [review]

**Pull request merged**